apps/src/codebridge/FileBrowser/prompts/openImportFromBackpackPrompt.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  NewFileFunction,
+  SaveFileFunction,
+} from '@codebridge/codebridgeContext/types';
+import {DEFAULT_FOLDER_ID} from '@codebridge/constants';
+import {validateFileName, getFileNameWithNumberSuffix} from '@codebridge/utils';
+
+import {MultiFileSource, ProjectFile} from '@cdo/apps/lab2/types';
+import {
+  DialogType,
+  DialogControlInterface,
+  extractUserInput,
+} from '@cdo/apps/lab2/views/dialogs';
+import {GenericDropdownProps} from '@cdo/apps/lab2/views/dialogs/GenericDropdown';
+import {BackpackContextType} from '@cdo/apps/sharedComponents/backpack/BackpackAPIContext';
+
+type OpenImportFromBackpackPromptArgsType = {
+  dialogControl: Pick<DialogControlInterface, 'showDialog'>;
+  backpackApi: BackpackContextType;
+  newFile: NewFileFunction;
+  saveFile: SaveFileFunction;
+  projectFiles: MultiFileSource['files'];
+  validationFile?: ProjectFile;
+};
+
+export const openImportFromBackpackPrompt = async ({
+  dialogControl,
+  backpackApi,
+  newFile,
+  saveFile,
+  projectFiles,
+  validationFile,
+}: OpenImportFromBackpackPromptArgsType) => {
+  const handleError = (message: string) => () => {
+    // TODO: send analytics / add logging.
+    console.error(message);
+  };
+  // Delete a file from the backpack.
+  const handleDelete = async (filename: string) => {
+    backpackApi.deleteFiles(
+      [filename],
+      handleError(`Error in deleting file ${filename}`),
+      () => console.log(`Deleted file ${filename}`)
+    );
+  };
+
+  // Fetch file content from backpack and then update or create a project file.
+  const fetchFileContentAndProcess = (
+    fileName: string,
+    newFileName?: string
+  ) => {
+    backpackApi.fetchFile(
+      fileName,
+      handleError(`Error in fetching file ${fileName}`),
+      (fileContent: string) => {
+        if (newFileName) {
+          newFile({fileName: newFileName, contents: fileContent});
+        } else {
+          const fileId = Object.keys(projectFiles).find(
```

Comment:
we should check for the file in a specific folder right? The user could theoretically have two of the same file name, in different folders. [This util](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/codebridge/utils/findFile.ts) may be helpful.

Reply:
Good point! I did look at this `util` function and was going to use, but the arg is a `MultiFileSource` instead of `MultiFileSource['files']` so opted to include the logic here but somehow missed the check for folder. Will update in this PR.